### DISCUSSION TITLE
v9.0.0

### DISCUSSION THREAD

**Discussion by tadashi-aikawa** [tadashi-aikawa/obsidian-another-quick-switcher#195] [Announcements]
### Backlink search (#177 ) `🏷️beta1`

The new Backlink search enables displaying all occurrences in the same file and listing the corresponding lines' text.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/9500018/bb7b937c-0a4d-488d-9047-479c8d9beb8c

### Link search (#177 ) `🏷️beta2`

The new Link search enables displaying all occurrences in the same file and listing the corresponding lines' text.

https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/9500018/6dbe4309-e078-4028-bdb4-9040759365cd

## ✨ **New things**

- Support Obsidian 1.4 and links in the properties `🏷️beta4`

## 🦾 **No longer broken**

- Fix a bug that shows -INFINITY as the fuzzy score `🏷️beta3`
- Fix a bug that the fuzzy score of the markdown file tends to be lower than the one of alias `🏷️beta5`
    - Fix a bug that the query doesn't hit an extension (other than .md) of the file `🏷️beta6`

**Comment by FelipeRearden** [comment]
> Backlink search (https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/issues/177 ) 🏷️beta1

Hello @tadashi-aikawa !!!!!

This is absolutely fantastic !!!! I have been playing with this new AQS Modal and it is exactly what I need :)

Thanks you very much for bring to us another amazing modal, you are super 🥇

Since we are in beta, **if you allow me,** I would like to share 4️⃣ 💡 ideas related to this new feature. 

## 💡1️⃣ Idea 1: New AQS Modal → Outlinks Search

The same approach created for `Backlinks Search` applied to **Outlinks**

→ displaying all internal links in active note and listing the corresponding lines text.

---

## 💡2️⃣ Idea 2: Backlink Dialog → Switch Between `Backlinks Search` and `Outlinks Search` and vice-versa

Only if 💡1️⃣ is approved 🙏

This Dialog command makes easy to switch between the two AQS Modals.

Since there is a strong relation between these two features, the idea here is to have a single hotkey command to open the AQS modal and then using the **Dialog Command** to switch between then 

---

## 💡3️⃣ Idea 3: Backlink Dialog → Filter for the Active `suggestion-item.is-selected`

This command would be useful when you 20/30 backlinks and more than 5 backlinks entries for each backlink note. 

The idea is to have a command to **filter** the AQS modal based in the Note that is selected in the AQS Modal

Following the example from your video:

1️⃣ The Suggestion item is *focused on the note*  `⚙️ Enable internal link complement` 

2️⃣ When hitting the new Backlink Dialog Command → it's will copy `⚙️ Enable internal link complement` to the `prompt.input` text field and automatic filter the AQS modal showing only the matches from `⚙️ Enable internal link complement`
📸 image from your video...
![EE3F8D84-0021-471D-BB31-813298BDAFF2](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/85180696/b8399042-bb5a-4713-b36c-7419454243f0)

3️⃣ hitting the new Backlink Dialog Command **AGAIN** will erase the `prompt.input` getting back to the standard view 


---

## 💡4️⃣ Idea 4: Backlink Dialog → Toggle to Show/Hide All `backlink items entries`

Again, this command would be useful when you 20/30 backlinks and more than 5 backlinks entries for each backlink note. 

When the number of backlinks is big, would be great to have an command to **hide the entries from these backlinks notes** to be able to see ALL backlinks and after that, toggle the command again to ,focus on a subset of notes that you want to see the backlinks entries. 


---
Forgive for the amount of new ideas, since this topic is important to me, I have a lot of ideas to bring to this new feature.

Been able to use AQS intelligence to navigate to backlinks is a game changer for my workflow. I have been struggling with the Core Plugin for a long time :)

Please feel free to discard any idea and let me know if you need more information about these ideas.

I wish you a fantastic day ☀️

**Reply by tadashi-aikawa** [reply]
Hi, @FelipeRearden. Thank you for your suggestions :)

> 💡1️⃣ Idea 1: New AQS Modal → Outlinks Search

Sorry, I couldn't think of a situation where this would be necessary. Could you tell me why 'Search target=link' is unsuitable in the "Custom Search"?

![image](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/9500018/92fa810b-ec29-44cc-b64e-ca680c341c8c)

> 💡3️⃣ Idea 3: Backlink Dialog → Filter for the Active suggestion-item.is-selected

Sorry, I couldn't think of a situation where this would be necessary. I'm concerned that adding commands of low necessity might complicate things.

**Reply by FelipeRearden** [reply]
Hello @tadashi-aikawa !!!!

Forgive me for my poor explanation, let me expand the concepts of this ideas to see if we get on the same page :)

> > 💡1️⃣ Idea 1: New AQS Modal → Outlinks Search
> 
> Sorry, I couldn't think of a situation where this would be necessary. Could you tell me why 'Search target=link' is unsuitable in the "Custom Search"?

📸 For me, the most important thing from the new AQS Modal  `Backlinks Search` is to be able to select/navigate **each line where the backlink exists**
![1636FC44-10D6-4C8A-B752-A8E491C83CFE](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/85180696/1e0818b1-ecdf-43b0-b4c8-26e14fd7d1b0)

💡1️⃣ Idea 1 is about bring the same degree of detail to the Outlinks/Links. As you can see from the image below, the Outlinks Core Plugin lacks this feature. 

📸 As you can see from the image below is this note has 3 matches/lines from each `Link` and the Core Plugin do not let me see or select any of then. I only shows to me that there are these links 
![E5DB7FDD-1AA2-4038-80FE-81AB89A80592](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/85180696/042489ea-3b0f-4f97-822b-ef0a3ab6d4cd)

📸As you know this is the same behavior of an Custom Search with 'Search target=link'
![41D9FBE4-9DDF-489E-AD32-DD968B98681F](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/85180696/7abe3612-3715-4d72-aa8d-a272b117ea70)

ℹ️ **The Difference Between AQS Modal Outlinks and Custom Search with 'Search target=link'**
→  Custom Search with 'Search target=link' will open the Note when we hit <kbd>ENTER</kbd>
→ AQS Modal Outlinks will **keep the Active Note open and transport/navigate to the line selected in the Modal** (it is the same behavior that we have for the Header Search modal applied to links); 

This type of detail is important when you have many references in a long Note. Many times I need to navigate/see what is the content of the Line that these `links` and not just know that the Active Note has `[[Lorem]]` and `[[ipsum]]` as `links`.
I would like to have the same power from `Backlinks` applied to `Links` in a AQS Modal 

It would be something like this:
```
**Lorem**
[1] 1️⃣ This is my first link from [[Lorem]] note
[2] 2️⃣ This is my second link from [[Lorem]] note
[3] 3️⃣ This is my third link from [[Lorem]] note

**ipsum**
[4] 🅰️ This is my first link from [[ipsum]] note
[5] 🅱️ This is my second link from [[ipsum]] note
[6] 🆎 This is my third link from [[ipsum]] note
```
 ---

---

>>💡3️⃣ Idea 3: Backlink Dialog → Filter for the Active suggestion-item.is-selected
>
> Sorry, I couldn't think of a situation where this would be necessary. I'm concerned that adding commands of low necessity might complicate things.

Let me explain ☀️

Image the the Note `Lorem` is linked with the following notes:
→ `ipsum` on 5 different lines;
→ `car` on 10 different lines;
→ `boat` on 20 different lines;

When you open the AQS Modal `Backlinks Search` from the Note `Lorem` it is gonna have a lot of information navigate since we have 35 items divided into 3 notes.

Many times I just wanna see the backlink lines from a **Specific Note**. This way, the users first decision after open the Modal is to filter the AQS Modal to focus on a list of backlink lines.
→ I just wanna see the 20 lines from the Note `boat` and take a decision from there...

💡3️⃣ Idea 3 solves this issue by filtering the content of the AQS Modal using a hotkey **without having to type the name of the note** just by using the `suggestion-item.is-selected`

I struggle with this a lot because my book notes has 3000/4000 lines and many **concepts/axioms/metal models notes** has 20 30 links on 3/5 different notes. 

When I open these **concepts/axioms/metal models notes** to search for connections using the Core Plugin Backlinks it is nightmare to navigate into that tiny box that does note have a good keyboard navigation.

Thats why I am asking for a filter in the AQS Modal, because some notes has a big workload :)

---

Please let me know if something is note clear from  💡3️⃣ and/or 💡1️⃣ 🙏

Thanks for the patience with my ideas @tadashi-aikawa ☀️

**Reply by tadashi-aikawa** [reply]
@FelipeRearden 
Thank you for the detail :)  I understood perfectly👍 

- I'll implement 1️⃣ 
- I'll consider the specification that fills the requirement of 3️⃣  (It doesn't fill the requirement in edge cases that the note title includes the title of the other)

**Reply by FelipeRearden** [reply]
Hello @tadashi-aikawa !!!!

> Thank you for the detail :) I understood perfectly👍

I am glad to know that we are on the same page about these ideas 😀

> I'll implement 1️⃣
Thank you very much @tadashi-aikawa 🎉🎉🎉

> I'll consider the specification that fills the requirement of 3️⃣ (It doesn't fill the requirement in edge cases that the note title includes the title of the other)
You 100% right @tadashi-aikawa , we need  a special attention to these edge cases.

If you allow me 2 more comments:
🅰️ Let me know if you want me to create the FR issues on `Issues`

🅱️ I kindly and humbly ask to consider 💡2️⃣ and 💡4️⃣ when organizing the new features 🙏🙏🙏

---

Thanks again @tadashi-aikawa , I wish you a fantastic day ☀️

**Reply by tadashi-aikawa** [reply]
Released 1️⃣  in [v9.0.0-beta2](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/9.0.0-beta2) 🚀

**Reply by FelipeRearden** [reply]
> Released 1️⃣ in [v9.0.0-beta2](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/releases/tag/9.0.0-beta2) 🚀

Amazing, Amazing, Amazing @tadashi-aikawa !!!!!!! 🎉
Working perfect on Obsidian Mobile version `1.3.4`  ✅

###💡 Just one silly idea about CSS to **Backlinks Modal** and **Links Modal**
If possible, would be great to have a specific class to the link/backlink `[[lorem ipsum]]` inside `.another-quick-switcher__link__item__description` / .another-quick-switcher__backlink__item__description` to make possível to add a different style thats highlights the link/backlink 

Just an idea :)

**Reply by tadashi-aikawa** [reply]
> If possible, would be great to have a specific class to the link/backlink [[lorem ipsum]] inside .another-quick-switcher__link__item__description / .another-quick-switcher__backlink__item__description` to make possível to add a different style thats highlights the link/backlink

It hasn't yet separated the DOM. We should add it in a later version :)

**Reply by FelipeRearden** [reply]
> It hasn't yet separated the DOM. We should add it in a later version 

Perfect @tadashi-aikawa !!!! Thank you very much!!!!!
